src/schroedinger.jl

```diff
@@ -102,3 +104,21 @@ function check_schroedinger(psi::Bra, H)
     check_multiplicable(psi, H)
     check_samebases(H)
 end
+
+
+_promote_time_and_state(tspan, psi0, f) = _promote_time_and_state(tspan, psi0, f(first(tspan), psi0))
+function _promote_time_and_state(tspan, psi0, H::AbstractOperator)
+    Ts = eltype(H)
+    Tt = real(Ts)
+    # promote only if ForwardDiff.Dual
+    if Tt <: ForwardDiff.Dual
```

Comment:
could you add an extra condition, that psi0 is not already dual? We do not want to apply `Dual` twice.

Reply:
done.
Also added check to promote if either state or time are `Dual `